src/App.js

```diff
@@ -0,0 +1,101 @@
+import { useEffect, useState } from 'react'
+import { Routes, Route, createSearchParams, useSearchParams, useNavigate } from "react-router-dom"
+import { useDispatch, useSelector } from 'react-redux'
+import 'reactjs-popup/dist/index.css'
+import { fetchMovies } from './data/moviesSlice'
+import { ENDPOINT_SEARCH, ENDPOINT_DISCOVER, ENDPOINT, API_KEY } from './constants'
+import Header from './components/Header'
+import Movies from './components/Movies'
+import Starred from './components/Starred'
+import WatchLater from './components/WatchLater'
+import YouTubePlayer from './components/YoutubePlayer'
+import './app.scss'
+
+const App = () => {
+
+  const state = useSelector((state) => state)
+  const { movies } = state  
+  const dispatch = useDispatch()
+  const [searchParams, setSearchParams] = useSearchParams()
+  const searchQuery = searchParams.get('search')
+  const [videoKey, setVideoKey] = useState()
+  const [isOpen, setOpen] = useState(false)
+  const navigate = useNavigate()
+  
+  const closeModal = () => setOpen(false)
+  
+  const closeCard = () => {
+
+  }
+
+  const getSearchResults = (query) => {
+    if (query !== '') {
+      dispatch(fetchMovies(`${ENDPOINT_SEARCH}&query=`+query))
+      setSearchParams(createSearchParams({ search: query }))
+    } else {
+      dispatch(fetchMovies(ENDPOINT_DISCOVER))
+      setSearchParams()
+    }
+  }
+
+  const searchMovies = (query) => {
+    navigate('/')
+    getSearchResults(query)
+  }
+
+  const getMovies = () => {
+    if (searchQuery) {
+        dispatch(fetchMovies(`${ENDPOINT_SEARCH}&query=`+searchQuery))
+    } else {
+        dispatch(fetchMovies(ENDPOINT_DISCOVER))
+    }
+  }
+
+  const viewTrailer = (movie) => {
+    getMovie(movie.id)
+    if (!videoKey) setOpen(true)
+    setOpen(true)
+  }
+
+  const getMovie = async (id) => {
+    const URL = `${ENDPOINT}/movie/${id}?api_key=${API_KEY}&append_to_response=videos`
+
+    setVideoKey(null)
+    const videoData = await fetch(URL)
+      .then((response) => response.json())
+
+    if (videoData.videos && videoData.videos.results.length) {
+      const trailer = videoData.videos.results.find(vid => vid.type === 'Trailer')
+      setVideoKey(trailer ? trailer.key : videoData.videos.results[0].key)
+    }
+  }
+
+  useEffect(() => {
+    getMovies()
+  }, [])
```

Comment:
I'm seeing this warning message: _React Hook useEffect has a missing dependency: 'getMovies'. Either include it or remove the dependency array._, so you should do this
```suggestion
  }, [getMovies])
```

**Additionally**

1. This list of movies is only being used in `Movies` component so the logic to get that data should be there and not here. Think about this scenario: the user goes directly to `watch-later` screen and reloads the page, with the pattern you're doing here you're requesting and getting the data, data that the user didn't ask for.
2. Avoid hasty abstraction, the function `getMovies` is only used here and the logic there is easy to understand and doesn't have many lines, so in my opinion, it's unnecessary to create a separate function.